src/renderers/shared/ReactDebugTool.js

```diff
@@ -192,6 +192,63 @@ function resumeCurrentLifeCycleTimer() {
   currentTimerType = timerType;
 }
 
+var lastMarkTimeStamp = null;
+var canUsePerformanceMeasure =
+  typeof performance !== 'undefined' &&
+  typeof performance.mark === 'function' &&
+  typeof performance.clearMarks === 'function' &&
+  typeof performance.measure === 'function' &&
+  typeof performance.clearMeasures === 'function';
+
+function shouldMark(debugID) {
+  if (!isProfiling || !canUsePerformanceMeasure) {
```

Comment:
Probably overkill but if `canUsePerformanceMeasure` is not true, I would just do `shouldMark = function() {}`.

You could use the same technique by clearing shouldMark when you end profiling and bring it back when you start profiling. This way you don't even have to do that test.

I have no idea what are the perf implications of this but since we're talking about super hot functions, it may be worth it.